components/ILIAS/Init/classes/class.ilStartUpGUI.php

```diff
@@ -127,6 +134,11 @@ public function getSafePostCommands(): array
     public function executeCommand(): void
     {
         $cmd = $this->ctrl->getCmd('processIndexPHP');
+        if (self::$forced_cmd) {
+            $cmd = self::$forced_cmd;
+            self::$forced_cmd = '';
+        }
```

Comment:
Could we maybe provide a method (or even better, a separate/tiny class, which encapsulates this check/set/reset sequence logic?